core/templates/pages/contributor-dashboard-admin-page/contributor-admin-dashboard-page.component.spec.ts

```diff
@@ -422,19 +414,140 @@ describe('Contributor dashboard Admin page', () => {
+      component.changeFirstActivity(
+        component.getDateThatIsDaysBeforeToday(numberOfDaysSinceFirstActivity)
+      );
+
+      expect(component.filter.numberOfDaysSinceLastActivityStartDate).toEqual(
+        numberOfDaysSinceFirstActivity
+      );
     }));
 
+    it(
+      "should not change filter by users' last activity when given start date " +
+        'is in the future',
+      fakeAsync(() => {
+        const numberOfDaysSinceFirstActivity = -1;
+        component.ngOnInit();
+        tick();
+        fixture.detectChanges();
+        const initialFirstActivity =
+          component.filter.numberOfDaysSinceLastActivityStartDate;
+
+        component.changeFirstActivity(
+          component.getDateThatIsDaysBeforeToday(numberOfDaysSinceFirstActivity)
+        );
+
+        expect(component.filter.numberOfDaysSinceLastActivityStartDate).toEqual(
+          initialFirstActivity
+        );
+      })
+    );
+
+    it(
+      "should not change filter by users' last activity when given start date " +
+        "comes before filter's current end date",
+      fakeAsync(() => {
+        component.ngOnInit();
+        tick();
+        fixture.detectChanges();
+        const initialFirstActivity =
+          component.filter.numberOfDaysSinceLastActivityStartDate;
+        const numberOfDaysSinceFirstActivity =
+          component.filter.numberOfDaysSinceLastActivityEndDate + 1;
+
+        component.changeFirstActivity(
+          component.getDateThatIsDaysBeforeToday(numberOfDaysSinceFirstActivity)
+        );
+
+        expect(component.filter.numberOfDaysSinceLastActivityStartDate).toEqual(
+          initialFirstActivity
+        );
+      })
+    );
+
+    it("should changes filter by users' last activity when end date changes", fakeAsync(() => {
+      const numberOfDaysSinceLastActivity = 30;
+      component.ngOnInit();
+      tick();
+      fixture.detectChanges();
+
+      component.changeLastActivity(
+        component.getDateThatIsDaysBeforeToday(numberOfDaysSinceLastActivity)
+      );
+
+      expect(component.filter.numberOfDaysSinceLastActivityEndDate).toEqual(
+        numberOfDaysSinceLastActivity
+      );
+    }));
+
+    it(
+      "should not change filter by users' last activity when given end date " +
+        'is in the future',
+      fakeAsync(() => {
+        const numberOfDaysSinceLastActivity = -1;
+        component.ngOnInit();
+        tick();
+        fixture.detectChanges();
+        const initialLastActivity =
+          component.filter.numberOfDaysSinceLastActivityEndDate;
+
+        component.changeLastActivity(
+          component.getDateThatIsDaysBeforeToday(numberOfDaysSinceLastActivity)
+        );
+
+        expect(component.filter.numberOfDaysSinceLastActivityEndDate).toEqual(
+          initialLastActivity
+        );
+      })
+    );
+
+    it(
+      "should not change filter by users' last activity when given end date " +
+        "comes after filter's current start date",
```

Comment:
Again, it should be `before`, not `after`.

Reply:
done